### ISSUE TITLE
Changing Text Input Wraps text in <p> Tags

### ISSUE THREAD

**Issue by suprafly** [yabwe/medium-editor#907]
I have two editable content divs which are being stacked vertically and the text in both of them is centered. When I edit the fields, even just adding one character, the whole text is wrapped in <p> tags. This interferes with my page's formatting.

If this is intended behaviour rather than an issue, please can someone point me to the function or line number where the <p> tags are being applied? I've looked and cannot find it.

Thanks.

**Comment by bhowerton** [comment]
Seeing the same paragraph tag issue, but it appears to be ok in 5.10.0, which is the version used in the demo.

https://yabwe.github.io/medium-editor/bower_components/medium-editor/dist/js/medium-editor.js

**Comment by nmielnik** [comment]
@suprafly are you using the latest version of medium-editor like what @bhowerton was pointing out?

**Comment by Kondra007** [comment]
Well, same problem here. Using the latest version of medium-editor.
Trying to make a simple admin page to post messages to Telegram channels.
Telegram API doesn't support `<p>` tags so I should use `disableReturn: true` to get rid of them. In this case, I can't make multuline text.

**Comment by nmielnik** [comment]
So @Kondra007 what I can tell you is that most browsers will handle inserting this `<p>` tag automatically, though exactly when it gets added can vary based on the browser you're using.

To fully achieve the functionality that you're describing, there would likely need to be a new extension or a new option added for medium-editor which would essentially disallow `<p>` tags and instead do something like:
* replace all `<p>` tags with a `<span>` + trailing `<br />`
* replace all `<p>` tags with another block element that is supported (ie `<div>`)

Essentially, the new code would need to listen for any changes to the editor, scan for `<p>` tags, and then update the content.

**Comment by Kondra007** [comment]
@nmielnik Thanks for quick reply! I'll try to play with code these days, maybe I can solve this problem myself.

**Comment by harzzn** [comment]
@nmielnik I believe you are talking about a different case than OP. @suprafly was referencing a situation where the text gets wrapped with p-tags with any key press. This is not caused by the browser and is instead a bug in medium-editor, previously reported in #829, #865, #149. And here is how to reproduce it and fix it.

### Reproduce

Use Chrome 48.0.2564.116 and Medium-editor 5.14.4

1.  Create a div that contains text and attach medium-editor to the div
`<div class="editable">Lorem ipsum dolor sit amet, consectetur adipiscing elit.<div>`
2.  Click on the text inside the div and press any non-special key. 
3.  Check the dom and you will see
`<div class="editable"><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.k</p><div>`

See a demo here: http://codepen.io/harzzn/pen/BKaZbR

### Fix

What I've gathered, this happens because always when the keyup event is fired medium-editor wraps the text in p-tags to achieve functionality described in #34. However, the wrapping should only occur when the element is previously empty and the element is not a block-container element. I've implemented a fix for this in https://github.com/html2wp/medium-editor/commit/994d3e15ac75f39c45d65496af9e5066f3c6a8dc by adding the two additional checks as described above. Does this seem reasonable?

**Comment by nmielnik** [comment]
Yeah @harzzn you're definitely right (apologies to @suprafly and @Kondra007), I just recently stumbled upon that functionality as part of the work for #986 and had forgotten that I commented about this before.

Thanks a ton for pointing this out, and for provided a potential fix. I need to wrap my head around the change a bit as this is a pretty critical code path but it seems to make sense.
